lightkurve/io.py

```diff
@@ -0,0 +1,243 @@
+        * `TessLightCurveFile` (typical suffix "_lc.fits").
+
+    Parameters
+    ----------
+    path_or_url : str
+        Path or URL of a FITS file.
+
+    Returns
+    -------
+    data : a subclass of  `~lightkurve.targetpixelfile.TargetPixelFile` or
+        `~lightkurve.lightcurvefile.LightCurveFile`, depending on the detected file type.
+
+    Raises
+    ------
+    ValueError : raised if the data product is not recognized as a Kepler or
+        TESS product.
+
+    Examples
+    --------
+    To read a target pixel file using its path or URL, simply use:
+
+        >>> tpf = read("mytpf.fits")  # doctest: +SKIP
+    """
+    log.debug("Opening {}.".format(path_or_url))
+    # pass header into `detect_filetype()`
+    try:
+        with fits.open(path_or_url) as temp:
+            filetype = detect_filetype(temp)
+            log.debug("Detected filetype: '{}'.".format(filetype))
+    except OSError as e:
+        filetype = None
+        # Raise an explicit FileNotFoundError if file not found
+        if 'No such file' in str(e):
+            raise e
+
+    # Community-provided science products
+    if filetype == "K2SFF":
+        return read_k2sff(path_or_url, **kwargs)
+    elif filetype == "EVEREST":
+        return read_everest(path_or_url, **kwargs)
+
+    # Official data products;
+    # if the filetype is recognized, instantiate a class of that name
+    if filetype is not None:
+        return getattr(__import__('lightkurve'), filetype)(path_or_url, **kwargs)
+    else:
+        # if these keywords don't exist, raise `ValueError`
+        raise ValueError("Not recognized as a Kepler or TESS data product: "
+                         "{}".format(path_or_url))
+
+
+def read_k2sff(path_or_url, ext="BESTAPER", **kwargs):
+    """Read a K2SFF light curve file.
+
+    More information: https://archive.stsci.edu/hlsp/k2sff
+
+    Parameters
+    ----------
+    path_or_url : str
+        Path or URL of a K2SFF light curve FITS file.
+    ext : str
+        Version of the light curve to use.  Valid options include "BESTAPER",
+        "CIRC_APER0" through "CIRC_APER9", and "PRF_APER0" through "PRF_APER9".
+
+    Returns
+    -------
+    lc : `KeplerLightCurve`
+        A populated light curve object.
+    """
+    f = fits.open(path_or_url)
+
+    # Raise an exception if the requested extension is invalid
+    validate_method(ext, supported_methods=[hdu.name.lower() for hdu in f])
+
+    args = {
+        "time": f[ext].data['T'],
+        "flux": f[ext].data['FCOR'],
+        "flux_unit": "",  # SFF light curves are normalized
+        "cadenceno": f[ext].data['CADENCENO'],
+        "targetid": f[0].header["KEPLERID"],
+        "channel": f[0].header["CHANNEL"],
+        "campaign": f[0].header["CAMPAIGN"],
+        "mission": f[0].header["MISSION"],
+        "ra": f[0].header["RA_OBJ"],
+        "dec": f[0].header["DEC_OBJ"],
+        "label": f[0].header["OBJECT"]
+    }
```

Comment:
Should we add something in this `lc` object to show it's not a pipeline product? e.g. changing the label to 
`{} [SFF]'.format(f[0].header["OBJECT"])`

Reply:
Agreed!  I'm implementing your suggestion.